src/server/transaction.cc

```diff
@@ -1477,14 +1481,22 @@ void Transaction::LogAutoJournalOnShard(EngineShard* shard) {
   if (cid_->IsWriteOnly() == 0 && (cid_->opt_mask() & CO::NO_KEY_TRANSACTIONAL) == 0)
     return;
 
-  // If autojournaling was disabled and not re-enabled, skip it
-  if ((cid_->opt_mask() & CO::NO_AUTOJOURNAL) && !renabled_auto_journal_.load(memory_order_relaxed))
-    return;
-
   auto journal = shard->journal();
   if (journal == nullptr)
     return;
 
+  if (result.status != OpStatus::OK) {
+    // We log NOOP even for NO_AUTOJOURNAL commands because the non-success status could have been
+    // due to OOM in a single shard, while other shards succeeded
+    journal->RecordEntry(txid_, journal::Op::NOOP, db_index_, unique_shard_cnt_,
```

Comment:
Check out JournalStreamer::Start.
It does not replicate NOOP, but instead today we use this to trigger journal writes to sink.  so this will not get to replica , did you run your tests with the flag  enable_multi_shard_sync set to true ? I think this will fail

Reply:
Nice catch.
How about I'll send a `PING`, but mark the opcode as `COMMAND` instead? Do you see any problem with that?

Reply:
btw it does pass with `--df enable_multi_shard_sync=true`

Reply:
I think PING is ok.
Oh I see that the test is using proactor_threads=1 so there is only one shard.
Lets change the test to run on 2 shards. you will need to increase the max memory size as well

Reply:
I still see the test with proactor_threads=1, can you change it ? see my comment above

Reply:
Sorry, I forgot about it :(
Done now.